tests/tracer/test_logging.py

```diff
@@ -0,0 +1,358 @@
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert "warning log" in content
+        assert re.search(LOG_PATTERN, content) is not None
+
+
+def test_debug_logs_streamhandler_default(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When DD_TRACE_DEBUG is true, debug logs are emitted to StreamHandler
+        following a configured logging.basicConfig and its defined format.
+        Note: When running ddtrace-run, the ddtrace-run logs still emit to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+logging.basicConfig(format='%(message)s')
+ddtrace_logger = logging.getLogger('ddtrace')
+
+assert ddtrace_logger.getEffectiveLevel() == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 0
+
+ddtrace_logger.warning('warning log')
+ddtrace_logger.debug('debug log')
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is None
+    assert b"warning log" in err
+    assert b"debug log" in err
+    assert out == b""
+
+    code = """
+import logging
+
+logging.basicConfig(format='%(message)s')
+ddtrace_logger = logging.getLogger('ddtrace')
+
+assert ddtrace_logger.getEffectiveLevel() == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 0
+
+ddtrace_logger.warning('warning log')
+ddtrace_logger.debug('debug log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is None
+    assert "program executable" in str(err)
+    assert b"warning log" in err
+    assert b"debug log" in err
+    assert out == b""
+
+
+def test_debug_logs_can_go_to_file_backup_count(
+    run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess, tmpdir
+):
+    """
+    When DD_TRACE_DEBUG is true and DD_TRACE_LOG_FILE has been specified, debug logs are
+        written to a file at the expected backup count.
+        Note: When running ddtrace-run, the ddtrace-run logs still emit to stderr.
+    """
+    env = os.environ.copy()
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "10"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.getEffectiveLevel() == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 10
+assert ddtrace_logger.handlers[0].backupCount == 1
```

Comment:
lets add a few debug logs to ensure that we always rotate log files in case the default logging behavior ever changes (which could impact this test)

Reply:
Added in the latest commit, with `ddtrace multiple debug log` right below this assertion.

Reply:
Marking this as resolved since this has been added.